### PR TITLE
Use published version for secp crate

### PR DESCRIPTION
**Pull request by grbIzl**
- Switch to the published version of parity-secp256k1 (that was required in order to publish parity-crypto)
- As I used the latest version of parity-secp256k1 I was able to switch this crate to rand-0.7.2

### CODE REVIEW

File: parity-crypto/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ required-features = ["publickey"]
 
 [dependencies]
 tiny-keccak = "1.4"
-eth-secp256k1 = { git = "https://github.com/paritytech/rust-secp256k1", rev = "a96ad75", optional = true }
+parity-secp256k1 = { version = "0.6.0", optional = true }
```
Comment (dvdplm) [inline parity-crypto/Cargo.toml]: This will likely be `0.7` when https://github.com/paritytech/rust-secp256k1/pull/29 lands.

Comment (grbIzl) [reply parity-crypto/Cargo.toml]: Ok, let me know, when you publish the new version, so I can proceed with this one.

Comment (grbIzl) [reply parity-crypto/Cargo.toml]: Changed

**Comment by ordian** [comment]
Please add a git tag `parity-crypto-v0.4.2` after publishing the crate.

**Pull request merged**